packages/at_client/lib/src/service/sync_service.dart

```diff
@@ -35,6 +35,7 @@ abstract class SyncService {
   /// Call this method to set the Global onDone callback.
   /// This method will be called when a sync is completed.
   /// When a specific onDone function is passed to the sync Function, Then the specific onDone is called.
+  @Deprecated('Use SyncProgressListener')
```

Comment:
Shouldn't we also deprecate the onDone parameter of the sync function?

Reply:
@gkc: The "onDone" callback is used sync_new_test.dart (sync unit tests) and deprecating in the sync method raised below caution when running the dart analyzer (there are 15 such issues reported by dart analyzer).
`
info • test/sync_new_test.dart:2577:24 • 'onDone' is deprecated and shouldn't be used. Use SyncProgressListener. Try replacing the use of the deprecated
          member with the replacement. • deprecated_member_use_from_same_package
`
Shall we refactor the tests to use SyncProgressListener and then deprecate the "onDone" in the sync method?

Reply:
Yes

Please also check all existing app source code in atsign-foundation for uses of onDone and raise tickets to change that code

Reply:
Sure Gary, Can the deprecating of "onDone" callback and refactoring of tests go in a separate PR?

Reply:
Yes, if you like? But why not do it now when you're already working in that area of the code?

If you do go ahead to merge this PR as-is, please first check all existing app source code in atsign-foundation for uses of onDone and raise tickets to change that code

Reply:
Refactored the tests to use SyncProgressListener and removed the usage of onDone callback.